### PR TITLE
base: u-boot-fio: upgrade 2022.04+fio to 2023.04+fio

### PR DESCRIPTION
**Pull request by MrCry0**
Ready to review.

Upgrade the mainline u-boot with fio patches to 2023.04. Fix machines configs synchronously to stay atomic.

### CODE REVIEW

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/qemuarm64/lmp.cfg [hunk 1]
```diff
@@ -11,3 +11,4 @@ CONFIG_PREBOOT="virtio scan; env load"
 # CONFIG_ENV_IS_IN_FLASH is not set
 # CONFIG_MTD is not set
 # CONFIG_MTD_NOR_FLASH is not set
+CONFIG_BOOTSTD=y
```
Comment (ricardosalveti) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/qemuarm64/lmp.cfg:14]: Do we actually need it?

Comment (MrCry0) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/qemuarm64/lmp.cfg:14]: It fixes building. Let me check what depends on it.

Comment (MrCry0) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/qemuarm64/lmp.cfg:14]: it fixes this
```
aarch64-linux-gnu-ld.bfd: boot/bootmeth_efi_mgr.o: in function `efi_mgr_check':
/home/cryo/work/u-boot/boot/bootmeth_efi_mgr.c:39: undefined reference to `bootflow_iter_check_system'
```

**Review by angolini — APPROVED** [review]

**Review by quaresmajose — APPROVED** [review]
LGTM

**Comment by ricardosalveti** [comment]
@igoropaniuk you tested this one on stm32 right?

**Pull request merged**